examples/Generic/Traverse.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Generic.Traverse (genericTraverseTests) where
+
+import qualified Control.Functor.Linear as Control
+import Data.Functor.Linear (genericTraverse)
+import qualified Data.Functor.Linear as Data
+import Generics.Linear.TH
+import Hedgehog
+import Prelude.Linear
+import Test.Tasty
+import Test.Tasty.Hedgehog (testProperty)
+import qualified Prelude
+
+data Pair a = MkPair a a
+  deriving (Show, Prelude.Eq)
+
+$(deriveGeneric1 ''Pair)
+
+data WithLog a = WithLog (Maybe String) a
+  deriving (Show, Prelude.Eq)
+
+instance Control.Functor WithLog where
+  fmap f (WithLog lx x) = WithLog (logApp Nothing lx) (f x)
```

Comment:
For what it's worth, this doesn't verify the `Functor` laws, because, for instance, `fmap id` is not the identity (as it get something logged). You can get away with it by asserting that the log string is not actually meaningful.

Honestly, though, I'd just use an applicative from linear-base.

Reply:
But we don't have many `Control.Applicative` instances in linear-base. I only see `Identity`, `IO`, `ReaderT`, `StateT`, `StateR`, and `(,) a`. Which one would make a good example in your opinion?

Reply:
`(,) a` is a pretty obvious candidate!

Reply:
Or `StateT Id`. But `(,) a` is likely to be the better choice.

Reply:
Sorry, I got confused for a moment. I thought you were looking for a `Traversable` instance to derive. For an appropriate `Applicative` for testing, is there such a thing as a free control applicative? Free structures tend to make good test cases.

Reply:
We don't have one defined. We probably could, but on the other hand, it could not be printed or compared for equality. I don't think it would actually make for a good test case.

The goal of this file is primarily to serve as a code example for documentation purposes. I think that using writer is fine. It's a perfectly serviceable applicative structure.